File: problems/arrays-hashing/valid-anagram/solution.test.ts
import { describe, test, expect } from 'vitest';
import { solutionFunction } from './solution';

describe('Valid Anagram', () => {
  test('Example 1', () => {
    const result = solutionFunction("anagram", "nagaram");
    expect(result).toBe(true);
  });

  test('Example 2', () => {
    const result = solutionFunction("rat", "car");
    expect(result).toBe(false);
  });

  test('Edge case: empty input', () => {
    const result = solutionFunction("", "");
    expect(result).toBe(true);
  });
});

File: problems/arrays-hashing/valid-anagram/solution_test.go
package valid_anagram

import "testing"

func TestSolutionFunction(t *testing.T) {
	tests := []struct {
		name     string
		strA     string
		strB     string
		expected bool
	}{
		{
			name:     "Example 1: anagram",
			strA:     "anagram",
			strB:     "nagaram",
			expected: true,
		},
		{
			name:     "Example 2: not anagram",
			strA:     "rat",
			strB:     "car",
			expected: false,
		},
		{
			name:     "Edge case: empty strings",
			strA:     "",
			strB:     "",
			expected: true,
		},
		{
			name:     "Edge case: different lengths",
			strA:     "a",
			strB:     "ab",
			expected: false,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := SolutionFunction(tt.strA, tt.strB)
			if result != tt.expected {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}